planner/mapper/targetable.go

```diff
@@ -86,6 +86,51 @@ func NewFilter() *Filter {
 	}
 }
 
+func (f *Filter) ToMap(mapping *core.DocumentMapping) map[string]any {
```

Comment:
todo: I think it is worth adding a quick comment to this function, as it is bit involved and the `any` part of the return type is not very descriptive.

Reply:
The return is a `map[string]any` which is the most common map type used in the entire code base for representing query related stuff. But I will add a docstring for the function

Reply:
> The return is a map[string]any which is the most common map type used in the entire code base for representing query related stuff.

I see that as making it almost worse though, as this type in 99% of other cases is a document, not a gql filter with a fairly limited shape.

Reply:
The same structured is used for parsing filter and order types, its used for rendering documents to their map, etc. Its usually pretty clear that its just a generic dictionary style map with string keys, and w.e values. This isn't any diff. +1 for the comment on the function, but the return type should be fairly clear as far as Go idiom is concerned.